### PR TITLE
WPF:WINDOWS Added ability to update application installed in Program Files folder

### PR DESCRIPTION
**Pull request by abodalevsky**
1) Desktop application can be installed at any location, the most popular are: 
 - _User local data folder_, in this case the bundle can be stored in the same location and be  unique for the user. 
 - _Program Files folder_, in this case the application is unique for the system and will be shared  among all users of the system, the bundle should be unique for the system as well. 

Usually, user has no write access to Program Files folder or at least admin privileges have to been requested.  In this case the bundle will be stored in ProgramData folder as it is recommended.

2) Small refactoring to share more code between UWP and WPF projects.

3) Windows examples were updated to `react-native-windows 0.43.0-rc.0`

### CODE REVIEW

**Comment by sergey-akhalkov** [comment]
Hi @abodalevsky, thank you so much for your contribution, LGMT.

File: windows/CodePush.Shared/CodePushUtils.cs [hunk 1]
```diff
@@ -61,13 +60,10 @@ internal static string GetAssetsBundlePrefix()
...
-        internal static string GetFileBundlePrefix()
+        internal static string ExtractSubFolder(string fullPath)
         {
-#if WINDOWS_UWP
-            return CodePushConstants.FileBundlePrefix;
-#else
-            return GetAppFolder();
-#endif
+            var codePushSubPathArray = fullPath.Split(Path.DirectorySeparatorChar);
+            return String.Join("/", codePushSubPathArray.SkipWhile((value, index) => codePushSubPathArray.Length - index > 4).ToArray());
```
Comment (sergey-akhalkov) [inline windows/CodePush.Shared/CodePushUtils.cs]: @abodalevsky, could you please clarify what means number 4?

File: windows/CodePush.Shared/CodePushConstants.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ internal class CodePushConstants
         internal const string UnzippedFolderName = "unzipped";
 #if WINDOWS_UWP
         internal const string AssetsBundlePrefix = "ms-appx:///ReactAssets/";
-        internal const string FileBundlePrefix = "ms-appdata:///local";
+        internal const string FileBundlePrefix = "ms-appdata:///local/";
```
Comment (sergey-akhalkov) [inline windows/CodePush.Shared/CodePushConstants.cs]: @abodalevsky, do I get it right if `FileBundlePrefix` const is used in `GetJavaScriptBundleFileAsync` method only, so there are no possible issues with this change?

File: windows/CodePush.Shared/CodePushUtils.cs (hunk 1 above)
Comment (abodalevsky) [inline windows/CodePush.Shared/CodePushUtils.cs]: 4 means last 4 segments of the path where JS bundle is located. The idea of the change was to unify the calculation of js bundle file location for UWP and WPF. While they have different prefix but last 4 segments are the same for both implementations.
For example for
 `F:\react-native-code-push\Examples\CodePushDemoApp\windows\CodePushDemoApp.Wpf\bin\x64\DebugBundle\CodePush\cc1dd1ab\CodePush\index.windows.bundle`
the part that will be added to specific prefix will be:
`CodePush/cc1dd1ab/CodePush/index.windows.bundle`

File: windows/CodePush.Shared/CodePushConstants.cs (hunk 2 above)
Comment (abodalevsky) [inline windows/CodePush.Shared/CodePushConstants.cs]: It should be fine I have checked it before commit

**Pull request merged**